Add apply_analytical! function (#532)
Ferrite-FEM/Ferrite.jl: docs/src/manual/boundary_conditions.md

        (x, t) -> ū + ∇ū  ⋅ (x - x̄)
    )
    ```

# Initial Conditions

When solving time-dependent problems, initial conditions, different from zero, may be required. 
For finite element formulations of ODE-type, 
i.e. ``\boldsymbol{x}'(t) = \boldsymbol{f}(\boldsymbol{x}(t),t)``, 
where ``\boldsymbol{x}(t)`` are the degrees of freedom,
initial conditions can be specified by the [`apply_analytical!`](@ref) function.
For example, specify the initial pressure as a function of the y-coordinate
```julia
ρ = 1000; g=9.81    # density [kg/m³] and gravity [N/kg]
grid = generate_grid(Quadrilateral, (10,10))
dh = DofHandler(grid); push!(dh, :u, 2); push!(dh, :p, 1); close!(dh)
x = zeros(ndofs(dh))
apply_analytical!(x, dh, :p, x->ρ*g*x[2])
```

See also [Time Dependent Problems](@ref) for one example. 

*Note about solving DAE:* 

> A Differential Algebraic Equations (DAE) is an equation of the form
> ``\boldsymbol{r}(\boldsymbol{x}(t),\boldsymbol{x}'(t),t)=\boldsymbol{0}``,
> which cannot be expressed as an ODE.

Reviewer: Maybe we can be a bit more precise here and give a quick example. My idea is as follows.

```suggestion
A Differential Algebraic Equations (DAE) is an equation of the form
``\boldsymbol{r}(\boldsymbol{x}(t),\boldsymbol{x}'(t),t)=\boldsymbol{0}``,
which usually cannot be expressed as a true ODE. They occur often,
but not always, in forms where some time derivatives are missing
``
x_1'(t) = f(\boldsymbol{x}(t),t)
0 = g(\boldsymbol{x}(t),t)`
``
```

In for such equations, it is usually necessary to specify initial conditions 
for both ``\boldsymbol{x}(0)`` and ``\boldsymbol{x}'(0)``, and these must be consistent,

> i.e. ``\boldsymbol{r}(\boldsymbol{x}(0),\boldsymbol{x}'(0),0)=\boldsymbol{0}``.

Reviewer: This is only true for index 1 DAEs. For index>1 you get hidden constraints which must be fulfilled too. This makes the initialization so tricky.

Author: Perhaps something like this then?
```suggestion
For such equations, it is usually necessary to specify initial conditions 
for both ``\boldsymbol{x}(0)`` and ``\boldsymbol{x}'(0)``. 
These functions must be consistent, i.e. fulfilling both ``\boldsymbol{r}(\boldsymbol{x}(0),\boldsymbol{x}'(0),0)=\boldsymbol{0}`` and all additional constraints.
```
(I think I'll change from `x` to `a` for the unknowns later, just to avoid confusion with the coordinate)

        (x, t) -> ū + ∇ū  ⋅ (x - x̄)
    )
    ```

# Initial Conditions

When solving time-dependent problems, initial conditions, different from zero, may be required. 
For finite element formulations of ODE-type, 
i.e. ``\boldsymbol{u}'(t) = \boldsymbol{f}(\boldsymbol{u}(t),t)``, 
where ``\boldsymbol{u}(t)`` are the degrees of freedom,
initial conditions can be specified by the [`apply_analytical!`](@ref) function.
For example, specify the initial pressure as a function of the y-coordinate
```julia

> ρ = 1000; g=9.81    # density [kg/m³] and gravity [N/kg]

Reviewer 2: ```suggestion
ρ = 1000; g = 9.81    # density [kg/m³] and gravity [N/kg]
```

grid = generate_grid(Quadrilateral, (10,10))
dh = DofHandler(grid); push!(dh, :u, 2); push!(dh, :p, 1); close!(dh)
u = zeros(ndofs(dh))

> apply_analytical!(u, dh, x->ρ*g*x[2], :p)

Reviewer 2: ```suggestion
apply_analytical!(u, dh, x -> ρ * g * x[2], :p)
```

```

See also [Time Dependent Problems](@ref) for one example. 

*Note about solving DAE:* 
A Differential Algebraic Equations (DAE) is an equation of the form
``\boldsymbol{r}(\boldsymbol{u}(t),\boldsymbol{u}'(t),t)=\boldsymbol{0}``,
which usually cannot be expressed as a true ODE. They occur often,
but not always, in forms where some time derivatives are missing

> ``

Reviewer 2: ````suggestion
```math
````

u_1'(t) = f(\boldsymbol{u}(t),t)
0 = g(\boldsymbol{u}(t),t)`

> ``

Reviewer 2: ````suggestion
```
````
